src/odemis/driver/smaract.py

```diff
@@ -1452,11 +1472,16 @@ def _doReference(self, future):
 
                 # The SA_MC references all axes at once. This function blocks
                 self.Reference()
+                logging.debug("Starting referencing")
                 # wait till reference completes
                 while True:
-                    ev = self.WaitForEvent(MC_5DOF_DLL.SA_MC_INFINITE)
+                    ev = self.WaitForEvent(100)
                     # check if move is done
                     if ev.type == MC_5DOF_DLL.SA_MC_EVENT_MOVEMENT_FINISHED:
+                        logging.debug("Referencing finished")
+                        if ev.i32 != MC_5DOF_DLL.SA_MC_OK:
+                            logging.error("Referencing failed with error %s", ev.i32)
+                            raise IOError("Referencing failed with error %s" % (ev.i32,))
```

Comment:
I've noticed that ev.i32 contains a "normal" error code. So it should be possible to raise a `SA_MCError`